pkg/traceql/engine_metrics.go

```diff
@@ -1189,10 +1243,32 @@ func (b *SimpleAdditionAggregator) Combine(in []*tempopb.TimeSeries) {
 	}
 }
 
-func (b *SimpleAdditionAggregator) Results() SeriesSet {
+func (b *SimpleAggregator) Results() SeriesSet {
 	return b.ss
 }
 
+// Simple addition aggregator. It adds existing values with the new sample. New indicates if the existing values were just initialized, ugly
+func simpleAdditionAggregator(_ bool, start, end, step uint64, existing *TimeSeries, samples []tempopb.Sample) {
+	for _, sample := range samples {
+		j := IntervalOfMs(sample.TimestampMs, start, end, step)
+		if j >= 0 && j < len(existing.Values) {
+			existing.Values[j] += sample.Value
+		}
+	}
+}
+
+// Simple min aggregator. It calculates the minumun between existing values and a new sample, New indicates if the existing values were just initialized, ugly
+func simpleMinAggregator(new bool, start, end, step uint64, existing *TimeSeries, samples []tempopb.Sample) {
+	for _, sample := range samples {
+		j := IntervalOfMs(sample.TimestampMs, start, end, step)
```

Comment:
I know this code is repeated. But I'm thinking on average where we will need to know the number of elements in the array

Reply:
$x =(m * x_a + n*x_b)/(m+n)$